src/ar_node_utils.erl

```diff
@@ -35,19 +35,26 @@ get_conflicting_txs(StateTXs, TX) ->
 			)
 	].
 
-%% @doc Get a specific full block (a block containing full txs) via
-%% blocks indep_hash.
+%% @doc Get a full block (a block containing all transactions) by the independent hash.
+%%      Try to find the block locally first. If we do not have the full block on disk, try to download it from peers.
 get_full_block(Peers, ID, BHL) when is_list(Peers) ->
-	% check locally first, if not found ask list of external peers for block
+	GetBlockFromPeersFun = fun() ->
+		get_full_block_from_remote_peers(ar_util:unique(Peers), ID, BHL)
```

Comment:
I see no reason why we don't store the block after fetching it. Do you?

Reply:
We are storing such blocks, but the calls are in `ar_node_worker` and `ar_fork_recovery`, where the blocks are processed.

Reply:
But `get_full_block/3` is using `ar_storage`, so it makes sense to me to also write to the storage. But the most important thing is that someone is writing it to storage. Let's just hope it stays like this...

Reply:
✅